tools/runner.py

```diff
@@ -0,0 +1,181 @@
+                        match_result = 0
+
+                state = state2
+            else:
+                done = True
+                if LEARNING:
+                    player1.learn(state, action1, reward1, state1, done1)
+
+                if reward1 != env.DRAW_REWARD:
+                    # player1 Won
+                    match_result = 1
+                else:
+                    # player1 Draw
+                    match_result = 0
+
+        # idx0 = Win, idx1 = Draw, idx2 = Loss
+        results[1 - match_result] += 1
+        print(f"{player1.name}:{player2.name}={results}")
+
+    return results
+
+
+def tournament_print(leaderboard):
+    for item in leaderboard:
+        leaderboard[item][4] = tournament_score(leaderboard[item])
+
+    leaderboard = sorted(leaderboard.items(), reverse=True, key=lambda itm: itm[1][4])
+
+    print("{:6s} {:25s} {:7s} | {:23s} | {:23s} | {:23s} | {:23s}".format("", "", "", "Starter Rounds", "Second Rounds", "Starter matches", "Second matches"))
+
+    print("{:6s} {:25s} {:7s} | {:7s} {:7s} {:7s} | {:7s} {:7s} {:7s} | {:7s} {:7s} {:7s} | {:7s} {:7s} {:7s}".format("Place", "Name", "Score", "Wins", "Draws", "Losses", "Wins", "Draws", "Losses", "Wins", "Draws", "Losses", "Wins", "Draws", "Losses"))
+    idx = 0
+    for item in leaderboard:
+        idx += 1
+        print(
+            "{:6d} {:25s} {:7d} | {:7d} {:7d} {:7d} | {:7d} {:7d} {:7d} | {:7d} {:7d} {:7d} | {:7d} {:7d} {:7d}".
+                format(idx, item[0], item[1][4],
+                       item[1][0][0], item[1][0][1], item[1][0][2],
+                       item[1][1][0], item[1][1][1], item[1][1][2],
+                       item[1][2][0], item[1][2][1], item[1][2][2],
+                       item[1][3][0], item[1][3][1], item[1][3][2]
+                       ))
+    pass
+
+
+def tournament_score(score):
+    # Score = Number of wins -  Number of losses
+    return (score[0][0] - score[0][2]) + (score[1][0] - score[1][2])
+
+
+def tournament(models):
+    players = []
+    for model in models:
+        players.append(tournament_player_loader(model))
+
+    print("Loaded", len(players), "players :", ", ".join([player.name for player in players]))
+
+    game_list = []
+    leaderboard = {}
+
+    for player1 in players:
+        leaderboard[player1.name] = [[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0], 0]
+        for player2 in players:
+            if player1 == player2:
+                continue
+            game_list.append((player1, player2))
+
+    random.shuffle(game_list)
+
+    for game in game_list:
+        score = play_game(game[0], game[1], ROUNDS)
+        print(score)
+
+        p1 = game[0].name
+        p2 = game[1].name
+        leaderboard[p1][0] = [x + y for x, y in zip(leaderboard[p1][0], score)]
+        leaderboard[p2][1] = [x + y for x, y in zip(leaderboard[p2][1], score[::-1])]
+        if score[0] > score[2]:
+            leaderboard[p1][2][0] += 1
+            leaderboard[p2][3][2] += 1
```

Comment:
Why increase score for both players? Shouldn't it decrease score for the player which lost?

Reply:
My bad, it's right.